### ISSUE TITLE
raster band value differs according to zoom/scale when using ascii and GeoTIFF versions of same raster

### ISSUE THREAD

**Issue by golfalot** [qgis/QGIS#45944]
### What is the bug or the crash?

**What fails**: "Identify Features" returns different band values from the GeoTiff depending how far zoomed out I am . When I zoom in close enough, band values from both layers are aligned. 

**What works**: "sample raster values" using half a dozen Points all return the same band value for both the ascii and GeoTiff.

### Steps to reproduce the issue

Documented in detail below with link to source files
https://gis.stackexchange.com/questions/415683/qgis-raster-band-value-differs-according-to-zoom-scale-when-using-ascii-and-geot

### Versions

QGIS 3.22.0-Białowieża

### Supported QGIS version

- [X] I'm running a supported QGIS version according to the roadmap.

### New profile

- [X] I tried with a new QGIS profile

**Comment by gioman** [comment]
> Using the raster calculator to subtract one raster from the other results in an output raster comprised entirely of zero values so there is no difference in your two rasters. Since neither of the two rasters have lower resolution overviews, it's probably just a slight difference in how the identify tool queries pixels at small scales (zoomed out)

@golfalot yeah, seems to be the case. Have you tried and see if changing this general option makes any difference?

![image](https://user-images.githubusercontent.com/1951107/140888411-91b44e77-f84d-49da-b2b1-2ef764181839.png)

**Comment by golfalot** [comment]
> > Using the raster calculator to subtract one raster from the other results in an output raster comprised entirely of zero values so there is no difference in your two rasters. Since neither of the two rasters have lower resolution overviews, it's probably just a slight difference in how the identify tool queries pixels at small scales (zoomed out)
> 
> @golfalot yeah, seems to be the case. Have you tried and see if changing this general option makes any difference?
> 
> ![image](https://user-images.githubusercontent.com/1951107/140888411-91b44e77-f84d-49da-b2b1-2ef764181839.png)

I tried some values between .01 to 100.0 but the inconsistent behaviour remains.
